### PR TITLE
libepoxy: set source charset to utf-8 fixing error C4819

### PR DESCRIPTION
**Pull request by taoyouh**
Specify library name and version:  **libepoxy/***

libepoxy considered C4819 as error (The file contains a character that cannot be represented in the current code page), and stops it from building under some locales (such as zh-CN with code page 936). This change made MSVC correctly read the UTF-8 sources and avoid the error.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — DISMISSED** [review]

File: recipes/libepoxy/all/conanfile.py [hunk 1]
```diff
@@ -90,6 +94,11 @@ def _configure_meson(self):
             defs[opt] = "true" if self.options.get_safe(opt, False) else "false"
         args=[]
         args.append("--wrap-mode=nofallback")
+
+        if self._is_msvc:
+            # Fix error C4819
+            args.append("-Dc_args=/source-charset:utf-8")
+
```
Comment (SpaceIm) [inline recipes/libepoxy/all/conanfile.py:101]: it seems to be fixed in master branch: https://github.com/anholt/libepoxy/blob/master/meson.build#L98

So maybe put this in a patch of meson.build instead?

Comment (taoyouh) [reply recipes/libepoxy/all/conanfile.py:101]: That sounds a good idea.

**Review by jgsogo — DISMISSED** [review]

File: recipes/libepoxy/all/conanfile.py [hunk 2]
```diff
@@ -40,6 +40,15 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    @property
+    def _is_msvc(self):
+        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+
```
Comment (SpaceIm) [inline recipes/libepoxy/all/conanfile.py:46]:
```suggestion
```

not required anymore

File: recipes/libepoxy/all/conanfile.py [hunk 3]
```diff
@@ -40,6 +40,15 @@ def _source_subfolder(self):
...
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
```
Comment (SpaceIm) [inline recipes/libepoxy/all/conanfile.py:48]:
```suggestion
```

**Review by SpaceIm — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**